### ISSUE TITLE
libassuan upgrade to 2.5.1 gets rid of at least one necessary lib file

### ISSUE THREAD

**Issue by jwhite007** [Linuxbrew/homebrew-core#5172]
- [x] Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Linuxbrew/brew/issues/new.
- [ ] Ran `brew update` and retried your prior step? NA
- [ ] Ran `brew doctor`, fixed all issues and retried your prior step? NA
- [ ] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link? NA
- [ ] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue? NA

To help us debug your issue please explain:

gpg-agent uses libassuan
```
$ ssh ct
gpg-agent: error while loading shared libraries: libassuan.so.0: cannot open shared object file: No such file or directory
```

```
$ ls Cellar/libassuan/2.5.1/lib/
libassuan.a
```

```
$ brew switch libassuan 2.4.4
Cleaning /home/linuxbrew/.linuxbrew/Cellar/libassuan/2.4.4
Cleaning /home/linuxbrew/.linuxbrew/Cellar/libassuan/2.5.1
8 links created for /home/linuxbrew/.linuxbrew/Cellar/libassuan/2.4.4
```
Expected:
```
$ ssh ct
Last login: Fri Dec 15 08:03:31 2017 from seadl0app001.ent.covance.com
gpg-agent: a gpg-agent is already running - not starting a new one
```

```
$ ls Cellar/libassuan/2.4.4/lib
libassuan.a  libassuan.so  libassuan.so.0  libassuan.so.0.7.4
```

**Comment by jwhite007** [comment]
```brew upgrade libassuan --build-from-source``` upgrades to 2.5.1 and includes the missing libassuan.so.0. With this I then thought that maybe 2.4.4 was built from source and that the latest release, 2.5.1, is bottled, but no....

```
$ brew info libassuan
libassuan: stable 2.5.1 (bottled)
Assuan IPC Library
https://www.gnupg.org/related_software/libassuan/
/home/linuxbrew/.linuxbrew/Cellar/libassuan/2.4.4 (16 files, 1.9MB)
  Poured from bottle on 2017-11-22 at 14:57:48
/home/linuxbrew/.linuxbrew/Cellar/libassuan/2.5.1 (16 files, 1.5MB) *
  Built from source on 2017-12-16 at 18:09:26
```

**Comment by iMichka** [comment]
Hi. I can confirm that the shared libraries (.so files) are missing from the 2.5.1 bottle. Last update was done 8 days ago #5124.
On mac the equivalent files are there (.dylib).

The build log tells us the following: https://circleci.com/api/v1.1/project/github/Linuxbrew/homebrew-core/10917/output/111/0?file=true

```
==> brew install --build-bottle --verbose libassuan
==> Downloading https://gnupg.org/ftp/gcrypt/libassuan/libassuan-2.5.1.tar.bz2
Already downloaded: /home/linuxbrew/.cache/Homebrew/libassuan-2.5.1.tar.bz2
==> Verifying libassuan-2.5.1.tar.bz2 checksum
tar xjf /home/linuxbrew/.cache/Homebrew/libassuan-2.5.1.tar.bz2
==> ./configure --disable-dependency-tracking --disable-silent-rules --prefix=/home/linuxbrew/.linuxbrew/Cellar/libassuan/2.5.1 --enable-static
.
==> make install
.
==> Cleaning
rm /home/linuxbrew/.linuxbrew/Cellar/libassuan/2.5.1/lib/libassuan.la
==> Finishing up
ln -s ../Cellar/libassuan/2.5.1/bin/libassuan-config libassuan-config
ln -s ../Cellar/libassuan/2.5.1/include/assuan.h assuan.h
ln -s ../../Cellar/libassuan/2.5.1/share/aclocal/libassuan.m4 libassuan.m4
ln -s ../../Cellar/libassuan/2.5.1/share/info/assuan.info assuan.info
info /home/linuxbrew/.linuxbrew/share/info/assuan.info
ln -s ../Cellar/libassuan/2.5.1/lib/libassuan.a libassuan.a
==> Not running post_install as we're building a bottle
You can run it manually using `brew postinstall libassuan`
==> Summary
/home/linuxbrew/.linuxbrew/Cellar/libassuan/2.5.1: 13 files, 1.4MB, built in 9 seconds
```

I will need to investigate further to see why this didn't build the .so files under Linux.

**Comment by iMichka** [comment]
We had the same problem with the libtiff bottle. a bottle rebuild should fix this. I’ll do that when I come back from work this evening.

**Comment by iMichka** [comment]
Fixed with #5270

**Comment by jwhite007** [comment]
@iMichka Thank you.
